### DISCUSSION TITLE
How to use read replicas in self-hosted supabase instance?

### DISCUSSION THREAD

**Discussion by thurahtetaung** [supabase/supabase#31003] [Questions]
I have a self-hosted supabase instance deployed in Kubernetes, with Cloudnative Postgres (CNPG) as the database. CNPG supports read-only endpoints and I would like to use read replicas on supabase when applicable. Currently, I am deploying the supabase services with configs very similar to the provided docker-compose so every service that need database access is only using the read/write endpoints. I would like guidance or documentation on how to integrate read replicas to them, as I have seen they are possible on supabase cloud so I think it should have an option to do similarly in self-hosted.

**Comment by styxlab** [comment]
@thurahtetaung Did you manage to set this up?

**Reply by thurahtetaung** [reply]
Unfortunately no, we gave up on that. There was not enough documentation available around self-hosting, especially on kubernetes.
